lib/teleterm/daemon/daemon.go

```diff
@@ -921,27 +922,45 @@ func (s *Service) ListUnifiedResources(ctx context.Context, clusterURI uri.Resou
 
 // GetUserPreferences returns the preferences for a given user.
 func (s *Service) GetUserPreferences(ctx context.Context, clusterURI uri.ResourceURI) (*api.UserPreferences, error) {
-	_, clusterClient, err := s.ResolveClusterURI(clusterURI)
+	_, rootClusterClient, err := s.ResolveClusterURI(clusterURI.GetRootClusterURI())
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	var leafClusterClient *client.TeleportClient
```

Comment:
`leafClusterClient` is used only for `leafClusterClient.SiteName`, isn't it? It can be extracted by calling `clusterURI.GetLeafClusterName`.